app/queries/outstanding_balance.rb

```diff
@@ -9,6 +9,10 @@
 #
 # See CompleteOrdersWithBalance or CustomersWithBalance as examples.
 class OutstandingBalance
+  # All the states of a finished order but that shouldn't count towards the balance (the customer
+  # didn't get the order for whatever reason). Note it does not include complete
+  FINALIZED_NON_SUCCESSFUL_STATES = %w(canceled returned).freeze
```

Comment:
I'm wondering about `awaiting_return` and `returned` now you've added them in the scope in `Spree::Order`. Are they relevant here? :thinking:

Reply:
For awaiting_return the customer is still in posession of the product, so until returned, we still need to consider it in the balance.